rundeckapp/grails-app/services/rundeck/services/ProjectService.groovy

```diff
@@ -912,8 +923,12 @@ class ProjectService implements InitializingBean, ExecutionFileProducer, EventPu
                         dir('executions/') {
                             //export executions
                             //export execution logs
+                            String remotePathTemplate = null
+                            if(execs.size() > 0)
+                                remotePathTemplate = logFileStorageService.getStorePathTemplateForExecution(execs.get(0))
+
                             execs.each { Execution exec ->
-                                exportExecution zip, exec, "execution-${exec.id}.xml"
+                                exportExecution zip, exec, "execution-${exec.id}.xml", remotePathTemplate
```

Comment:
It obtains the configured remote storage path just once with the first execution for efficiency since the configured path for log storage exists at framework level being the same for all executions and getting this property value requires instancing a LogFileStoragePlugin object.

On a project with **13656** executions (all logs in remote storage), exporting all the executions only and measuring the time since pressing the _Export Archive_ button, until the archive is available to be saved.

* AVG time for getting the expanded path for each execution: **1' 16,28"**

* AVG time for getting the remotePathTemplate just once and expand it for each execution (chosen implementation): **0' 36.53"**